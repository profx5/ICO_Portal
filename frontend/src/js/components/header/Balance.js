import React from 'react';


export default class Balance extends React.Component {
    constructor(props) {
        super(props);

    }

    render () {
        return (
            <div className="Header_balance col-md-3">
                <h3 className="Header_head">On-Chain balance</h3>
                <p className="Header_text Header_text-big">{this.props.currentAmount}</p>
                <button className="btn btn-primary">CONNECT WALLET</button>
            </div>
        );
    }
}