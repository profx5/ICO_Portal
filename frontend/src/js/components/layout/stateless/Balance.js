import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {media} from 'js/utils/media';

import walletIcon from 'img/icons/icon_wallet.svg';


const Balance = ({amount}) => {
    return (
        <Wrapper>
            <p className="Balance_head">Balance</p>
            <div className="Balance_amountWrapper">
                <img alt="Wallet icon" src={walletIcon}/>
                <p className="Balance_amount">{(amount / 10 ** 18).toFixed(2) || 0.00} <span>OGD</span></p>
            </div>
        </Wrapper>
    )
}

Balance.propTypes = {
    amount: PropTypes.number
}


export default Balance;

const Wrapper = styled.div`
    position: relative;
    padding: 0 23px;
    ${media.sm} {
        padding-right: 0;
    }
    ${media.xs} {
        flex-wrap: wrap;
    }
    img {
        margin-right: 15px;
        ${media.xs} {
            height: 14px;
            width: auto;
            margin-right: 8px;
        }
    }
    a {
        color: #3172fd;
        font-weight: 600;
        font-size: 20px;
    }
    span {
        font-weight: 400;
    }
    .Balance_head {
        ${media.xs} {
            font-size: 12px;
            display: block;
            text-align: right;
            margin-bottom: 6px;
        }
        ${media.smPlus} {
            display: none;
        }
    }
    .Balance_amountWrapper {
        white-space: nowrap;
        display: flex;
        align-items: center;
    }
    .Balance_amount {
        color: #3172fd;
        ${media.xs} {
            font-size: 14px;
            white-space: nowrap;
        }   
    }
`;
