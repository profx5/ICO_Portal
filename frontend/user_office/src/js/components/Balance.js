import React from 'react';
//components
import Button from './Button'
import Title from './Title'

const Balance = ({currentAmount, investClick}) => (
    <div className="Header_balance col-md-3">
        <Title type='h3' text='On-Chain balance'/>
        <p className="Header_text Header_text-big">{currentAmount}</p>
        <Button primary={true} text='CONNECT WALLET' onClick={investClick}/>
    </div>
)

export default Balance
