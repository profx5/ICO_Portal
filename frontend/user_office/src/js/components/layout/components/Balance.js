import React from 'react';
import styled from 'styled-components';

import walletIcon from './../../../../img/icons/icon_wallet.svg';


const Balance = ({amount}) => {
    return (
        <Wrapper>
            <img alt="Wallet icon" src={walletIcon}/>
            <p>{(amount / 10 ** 18).toFixed(2)} <span>OGD</span></p>
        </Wrapper>
    )
}


export default Balance;

const Wrapper = styled.div`
    position: relative;
    display: flex;
    align-items: center;
    padding: 0 23px;
    img {
        margin-right: 15px;
    }
    a {
        color: #3172fd;
        font-weight: 600;
        font-size: 20px;
    }
    span {
        font-weight: 400;
    }
`;
