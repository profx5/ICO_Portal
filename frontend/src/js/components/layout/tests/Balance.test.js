import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Balance from 'js/components/layout/stateless/Balance';


configure({ adapter: new Adapter() });

it('should render stateless Balance', () => {
    const component = shallow(<Balance/>);
    expect(component).toMatchSnapshot();
})
