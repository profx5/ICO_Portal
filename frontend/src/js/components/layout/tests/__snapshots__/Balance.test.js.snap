// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render stateless Balance 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Balance />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <p
          className="Balance_head"
        >
          Balance
        </p>,
        <div
          className="Balance_amountWrapper"
        >
          <img
            alt="Wallet icon"
            src="icon_wallet.svg"
          />
          <p
            className="Balance_amount"
          >
            NaN
             
            <span>
              OGD
            </span>
          </p>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Balance",
          "className": "Balance_head",
        },
        "ref": null,
        "rendered": "Balance",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Wallet icon"
              src="icon_wallet.svg"
            />,
            <p
              className="Balance_amount"
            >
              NaN
               
              <span>
                OGD
              </span>
            </p>,
          ],
          "className": "Balance_amountWrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Wallet icon",
              "src": "icon_wallet.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "NaN",
                " ",
                <span>
                  OGD
                </span>,
              ],
              "className": "Balance_amount",
            },
            "ref": null,
            "rendered": Array [
              "NaN",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "OGD",
                },
                "ref": null,
                "rendered": "OGD",
                "type": "span",
              },
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    position: relative;
    padding: 0 23px;
    ",
          "@media (min-width: 768px) and (max-width: 1024px)",
          " {
        padding-right: 0;
    }
    ",
          "@media (max-width: 767px)",
          " {
        flex-wrap: wrap;
    }
    img {
        margin-right: 15px;
        ",
          "@media (max-width: 767px)",
          " {
            height: 14px;
            width: auto;
            margin-right: 8px;
        }
    }
    a {
        color: #3172fd;
        font-weight: 600;
        font-size: 20px;
    }
    span {
        font-weight: 400;
    }
    .Balance_head {
        ",
          "@media (max-width: 767px)",
          " {
            font-size: 12px;
            display: block;
            text-align: right;
            margin-bottom: 6px;
        }
        ",
          "@media (min-width: 768px)",
          " {
            display: none;
        }
    }
    .Balance_amountWrapper {
        white-space: nowrap;
        display: flex;
        align-items: center;
    }
    .Balance_amount {
        color: #3172fd;
        ",
          "@media (max-width: 767px)",
          " {
            font-size: 14px;
            white-space: nowrap;
        }   
    }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <p
            className="Balance_head"
          >
            Balance
          </p>,
          <div
            className="Balance_amountWrapper"
          >
            <img
              alt="Wallet icon"
              src="icon_wallet.svg"
            />
            <p
              className="Balance_amount"
            >
              NaN
               
              <span>
                OGD
              </span>
            </p>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Balance",
            "className": "Balance_head",
          },
          "ref": null,
          "rendered": "Balance",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="Wallet icon"
                src="icon_wallet.svg"
              />,
              <p
                className="Balance_amount"
              >
                NaN
                 
                <span>
                  OGD
                </span>
              </p>,
            ],
            "className": "Balance_amountWrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Wallet icon",
                "src": "icon_wallet.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "NaN",
                  " ",
                  <span>
                    OGD
                  </span>,
                ],
                "className": "Balance_amount",
              },
              "ref": null,
              "rendered": Array [
                "NaN",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "OGD",
                  },
                  "ref": null,
                  "rendered": "OGD",
                  "type": "span",
                },
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    position: relative;
    padding: 0 23px;
    ",
            "@media (min-width: 768px) and (max-width: 1024px)",
            " {
        padding-right: 0;
    }
    ",
            "@media (max-width: 767px)",
            " {
        flex-wrap: wrap;
    }
    img {
        margin-right: 15px;
        ",
            "@media (max-width: 767px)",
            " {
            height: 14px;
            width: auto;
            margin-right: 8px;
        }
    }
    a {
        color: #3172fd;
        font-weight: 600;
        font-size: 20px;
    }
    span {
        font-weight: 400;
    }
    .Balance_head {
        ",
            "@media (max-width: 767px)",
            " {
            font-size: 12px;
            display: block;
            text-align: right;
            margin-bottom: 6px;
        }
        ",
            "@media (min-width: 768px)",
            " {
            display: none;
        }
    }
    .Balance_amountWrapper {
        white-space: nowrap;
        display: flex;
        align-items: center;
    }
    .Balance_amount {
        color: #3172fd;
        ",
            "@media (max-width: 767px)",
            " {
            font-size: 14px;
            white-space: nowrap;
        }   
    }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
